### DISCUSSION TITLE
Serving Files From wwwroot

### DISCUSSION THREAD

**Discussion by websitewill** [OrchardCMS/OrchardCore#15770] [Q&A]
I am still trying to sort out some silly Blazor hosting madness. Can someone tell me why in the world files that exist in wwwroot can't be served for a path but a files that DOES NOT EXIST can be served?
This blazor stuff creates a file called blazor.web.js and serves it from the root directory. The browser can find this file just fine.
But ALL of the other files that end us in my wwwroot folder when I publish the site? Nope - those get a 500 error (this specific error):
"Please register the EndpointMiddleware using 'IApplicationBuilder.UseEndpoints(...)' if using routing."

Here is the entirety of my Startup.cs file. Please help me make sense of this...
`
using System;
 //Other usings omitted for brevity

namespace XYZ.Client.CustomerPortal
{
    public class Startup
    {
        public IConfigurationRoot Configuration { get; set; }

        public Startup()
        {
            var configurationBuilder = new ConfigurationBuilder()
                .AddEnvironmentVariables()
                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true);

            Configuration = configurationBuilder.Build();
        }

        public void ConfigureServices(IServiceCollection services)
        {
            services.Configure<IdentityOptions>(options => { Configuration.GetSection("IdentityOptions").Bind(options); });

            //NOTE This cannot exist in the AddOrchardCms section, it has to be at this level.
            //This has to exist in the web project Startup, otherwise server-side blazor components won't work.
            services.AddRazorComponents()
                .AddInteractiveServerComponents()
                .AddInteractiveWebAssemblyComponents();
            
            //services.AddScoped<IClaimsTransformation, UserInfoClaims>();
            services.AddCors(
                options => options.AddPolicy("AllowCors", builder =>
                {
                    builder
                        .AllowAnyOrigin()
                        .AllowCredentials()
                        .AllowAnyHeader()
                        .AllowAnyMethod();
                })
            );
            
            services.AddSignalR();

            services.AddOrchardCms()
                .AddSetupFeatures("OrchardCore.AutoSetup")
                .ConfigureServices(services =>
                {
                    services.AddHttpsRedirection(options => { options.HttpsPort = 443; });
                })
                .Configure((app, routes) =>
                {
                    app.UseHttpsRedirection();
                    app.UseBlazorFrameworkFiles();

                    app.UseRouting();
                    app.UseAuthentication();
                    app.UseAuthorization();
                    app.UseAntiforgery();
                });
        }

        public void Configure(IApplicationBuilder app,  IWebHostEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
                app.UseWebAssemblyDebugging();
            }
            else
            {
                app.UseExceptionHandler("/Error");
                app.UseHsts();
            }

            app.UseOrchardCore();
        }
    }
}
`

What is the special order or parameter or <whatever> I am missing to get these to be available?

TIA,
Will

**Comment by websitewill** [comment]
@ns8482e 
I came across this. 
https://github.com/OrchardCMS/OrchardCore/issues/6505

That UseStaticFiles you mentioned doesn't seem to be working for me. I've tried it in the above along with UseDefaultFiles and other things that look like they should work (based on the description).
I've researched the order (before/after UseRouting, UseAuthZ, etc). No luck.

Any ideas?

**Comment by ns8482e** [comment]
In your code above, add UseStaticFiles() before app.UseBlazorFrameworkFiles();
